python/ray/dag/tests/experimental/test_accelerated_dag.py

```diff
@@ -136,18 +144,188 @@ def test_basic(ray_start_regular):
+        o3 = b.echo.bind(o1)
+        o4 = b.echo.bind(o2)
+        dag = MultiOutputNode([o3, o4])
 
-    with pytest.raises(
-        ValueError,
-        match="Compiled DAGs only supports actor methods with " "num_returns=1",
-    ):
-        dag.experimental_compile()
+    res = ray.get(dag.execute(1))
+    assert res == [1, 2]
+
+    compiled_dag = dag.experimental_compile()
+    res = ray.get(compiled_dag.execute(1))
+    assert res == [1, 2]
+
+    compiled_dag.teardown()
+
+
+def test_two_returns_one_reader_multi_times(ray_start_regular):
+    a = Actor.remote(0)
+    b = Actor.remote(0)
+    with InputNode() as i:
+        o1, o2 = a.return_two.bind(i)
+        o3 = b.echo.bind(o1)
+        o4 = b.echo.bind(o2)
+        dag = MultiOutputNode([o3, o4])
+
+    for _ in range(3):
+        res = ray.get(dag.execute(1))
+        assert res == [1, 2]
+
+    compiled_dag = dag.experimental_compile()
+    for _ in range(3):
+        res = ray.get(compiled_dag.execute(1))
+        assert res == [1, 2]
+
+    compiled_dag.teardown()
+
+
+def test_two_returns_two_readers(ray_start_regular):
+    a = Actor.remote(0)
+    b = Actor.remote(0)
+    c = Actor.remote(0)
+    with InputNode() as i:
+        o1, o2 = a.return_two.bind(i)
+        o3 = b.echo.bind(o1)
+        o4 = c.echo.bind(o2)
+        dag = MultiOutputNode([o3, o4])
+
+    res = ray.get(dag.execute(1))
+    assert res == [1, 2]
+
+    compiled_dag = dag.experimental_compile()
+    res = ray.get(compiled_dag.execute(1))
+    assert res == [1, 2]
+
+    compiled_dag.teardown()
+
+
+def test_two_returns_two_readers_multi_times(ray_start_regular):
+    a = Actor.remote(0)
+    b = Actor.remote(0)
+    c = Actor.remote(0)
+    with InputNode() as i:
+        o1, o2 = a.return_two.bind(i)
+        o3 = b.echo.bind(o1)
+        o4 = c.echo.bind(o2)
+        dag = MultiOutputNode([o3, o4])
+
+    for _ in range(3):
+        res = ray.get(dag.execute(1))
+        assert res == [1, 2]
+
+    compiled_dag = dag.experimental_compile()
+    for _ in range(3):
+        res = ray.get(compiled_dag.execute(1))
+        assert res == [1, 2]
+
+    compiled_dag.teardown()
+
+
+def test_two_as_one_return(ray_start_regular):
+    a = Actor.remote(0)
+    b = Actor.remote(0)
+    with InputNode() as i:
+        o1 = a.return_two_as_one.bind(i)
+        o2 = b.echo.bind(o1)
+        dag = o2
+
+    compiled_dag = dag.experimental_compile()
+    for _ in range(3):
+        res = ray.get(compiled_dag.execute(1))
+        assert res == (1, 2)
+
+    compiled_dag.teardown()
+
+
+def test_two_from_three_returns(ray_start_regular):
+    a = Actor.remote(0)
+    b = Actor.remote(0)
+    c = Actor.remote(0)
+    with InputNode() as i:
+        o1, o2 = a.return_two_from_three.bind(i)
+        o3 = b.echo.bind(o1)
+        o4 = c.echo.bind(o2)
+        dag = MultiOutputNode([o3, o4])
+
+    compiled_dag = dag.experimental_compile()
+    for _ in range(3):
+        res = ray.get(compiled_dag.execute(1))
```

Comment:
This should throw an error because the number of values returned by the task does not match the annotated number of returns. You can take a look at the error that this throws without `experimental_compile`, if you just directly call `a.return_two_from_three.remote(1)`

Reply:
Ack. Done.